packages/dds/tree/src/rebase/rebaser.ts

```diff
@@ -166,14 +166,23 @@ export interface ChangeRebaser<TChangeset> {
      * - `rebase(a, compose([]))` is equal to `a`.
      * - `rebase(compose([]), a)` is equal to `a`.
      */
-    rebase(change: TChangeset, over: TChangeset): TChangeset;
+    rebase(change: TChangeset, over: TaggedChange<TChangeset>): TChangeset;
 
     // TODO: we are forcing a single AnchorSet implementation, but also making ChangeRebaser deal depend on/use it.
     // This isn't ideal, but it might be fine?
     // Performance and implications for custom Anchor types (ex: Place anchors) aren't clear.
     rebaseAnchors(anchors: AnchorSet, over: TChangeset): void;
 }
 
+export interface TaggedChange<TChangeset> {
+    revision: RevisionTag | undefined;
+    change: TChangeset;
```

Comment:
We may as well make those readonly
```suggestion
    readonly revision: RevisionTag | undefined;
    readonly change: TChangeset;
```